Client/core/Graphics/CGraphics.cpp

```diff
@@ -488,6 +488,9 @@ void CGraphics::CheckModes(EDrawModeType newDrawMode, EBlendModeType newBlendMod
     // Draw mode changing?
     if (bDrawModeChanging || bBlendModeChanging)
     {
+        SetBlendModeRenderStates(newBlendMode);

```

Comment:
Why did you move `SetBlendModeRenderStates` to here?

Reply:
The first shader or primitive in the queue didn't change the blend mode for some reason.

Reply:
Changing the render states here will mean buffered draws will use the incorrect blend mode. (The following `->End()` and `->Flush()` calls ensure rendering using the previous blend mode is complete)

I think the first shader problem is caused by something else